### PR TITLE
Make plan parsing a little more robust.

### PR DESCRIPTION
**Pull request by raskchanky**
Closes https://github.com/habitat-sh/builder/issues/368
Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/core/src/package/plan.rs [hunk 1]
```diff
@@ -30,11 +30,27 @@ impl Plan {
...
+                // have a parser for (LUA!), but both of those things are beyond the
+                // scope of this task.
+                let parts: Vec<&str> = line.splitn(2, "=").map(|x| x.trim()).collect();
+
+                if parts.len() != 2 {
+                    continue;
+                }
+
+                let mut val = parts[1].replace("\"", "");
+                val = val.replace("'", "");
```
Comment (stevendanna) [inline components/core/src/package/plan.rs]: This implies that the fields being parsed cannot contain `'` or `"` (i.e. no `pkg_name="jim's-nice-service"`). I think in practice this is fine.

**Comment by baumanj** [comment]
Is there a good reason not to leverage bash itself for this? If you direct the bytes into `bash` but `cat` on a script like this:
```bash
#!/bin/bash

echo "$pkg_name"
echo "$pkg_origin"
echo "$pkg_version"
```
you'd get clean, predictable output in the way that `bash` itself would parse it. For example:
```
➤ cat /Users/jbauman/core-plans/wordpress/plan.sh ../parse-plan.sh | /bin/bash ^/dev/null
wordpress
core
4.7.4
```

**Comment by christophermaier** [comment]
@baumanj I think that opens us up to remote code execution:

`plan.sh`:
```bash
echo "HACKED"
pkg_name="my_package"
# etc...
```

`test.sh`: 
``` bash
#!/bin/bash
echo $pkg_name
```
Execution: 
```bash
cat plan.sh test.sh | bash
HACKED
my_package
```

**Comment by elliott-davis** [comment]
@baumanj @cm we have a similar tool here https://github.com/habitat-sh/core-plans/blob/master/bin/print-deps.sh

**Comment by baumanj** [comment]
Good point, @christophermaier. That raises the question: why are we keeping what is clearly *data*, that needs to be accessed by other parts of the system mixed in with executable code? Have we considered splitting this out?

Another way this bites us is that currently a plan can do runtime things to calculate these values (including sourcing other files). That will work in some parts of our system, but not others. I think it may be smart for long term security and usability to separate immutable data from executable logic in the plan.

**Pull request merged**